src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -242,14 +251,25 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
             }
         }
 
-        private static async Task ConsumeStreamReaderAsync(StreamReader reader, StringBuilder lines)
+        private static bool IsIgnoredOutput(string line, string[] excluded)
+        {
+            return excluded.All(p => line.ToLower().Contains(p));
```

Comment:
I would use a comparer instead of a making a new lowercase string to reduce memory usage

``line.Contains(p, StringComparer.OrdinalIgnoreCase)``